src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt

```diff
@@ -85,13 +88,26 @@ class SnykLanguageClient :
         Caffeine.newBuilder().expireAfterWrite(10, TimeUnit.SECONDS).build()
     private val progressEndMsgCache: Cache<String, WorkDoneProgressEnd> =
         Caffeine.newBuilder().expireAfterWrite(10, TimeUnit.SECONDS).build()
+    private var diagnosticsMap: ConcurrentHashMap<String, List<Diagnostic>> =
```

Comment:
This looks like a duplicate. It would be better to send the diagnostics to the cache via the event via the event bus directly.